### PR TITLE
Refactor scrapystrap to not be a single file

### PR DESCRIPTION
**Pull request by dturn**
@dalehamel @dwradcliffe Thoughts on where the refactor is going?

### CODE REVIEW

File: lib/simple_http.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+require 'net/http'
+require 'uri'
+require 'json'
+
+class SimpleHTTP
```
Comment (dalehamel) [inline lib/simple_http.rb]: we should replace this with faraday or something more standard, I this was just for when this was a single file script.

**Comment by dalehamel** [comment]
:+1: looking good, it would be neat to see helpers like 'sends_to_zygote' in here, and I think it makes sense to make a 'zygote client' instead of raw HTTP requests.

**Comment by dwradcliffe** [comment]
first glance looks good!

File: lib/task_base.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require './lib/utils'
+
+class TaskBase
```
Comment (dwradcliffe) [inline lib/task_base.rb]: Maybe `ActionHandlerBase` since that's the terminology you used?

**Comment by dwradcliffe** [comment]
Can you setup travis ci?

**Comment by dalehamel** [comment]
> Can you setup travis ci?

:+1: to that, and add a badge to the readme and make it a pre-merge check in github please :)

**Comment by dalehamel** [comment]
now that this isn't scrappy anymore, we also need a new name.

**Comment by dalehamel** [comment]
```genesis_handler``` ?

File: lib/simple_http.rb (hunk 1 above)
Comment (dturn) [inline lib/simple_http.rb]: How would you feel about doing that in a second PR? This one is going to get big w/ the new tests and I think there is something to be said for only moving code around in the first pass. Then changing code in a second pass.

Comment (dwradcliffe) [inline lib/simple_http.rb]: I'd prefer it in a 2nd PR too.

Comment (dalehamel) [inline lib/simple_http.rb]: seems like a sane approach to keep them separate, yeah

**Comment by dturn** [comment]
@Shopify/metalscale at 500 lines I think this is a good time to stop adding to this PR and have it reviewed

**Pull request merged**